### PR TITLE
Document fluent route changes.

### PR DESCRIPTION
**Pull request by markstory**
Update the documentation for the new fluent methods on Route. I've also broken the new features section up by package as it was becoming a wall of un-scanable text.

I've tried to follow the latest decisions on documenting the most recent version, and including previous version compatibility in follow up examples or in comments.

### CODE REVIEW

File: en/development/routing.rst [hunk 1]
```diff
@@ -338,6 +396,44 @@ CakePHP, and should not be used unless you want the special meaning
...
+
+    // Only match on the blog subdomain.
+    ->setHost('blog.example.com')
+
+    // Set the route elements that should be converted to passed arguments
+    ->setPass(['slug'])
+
+    // Set the matching patterns for route elements
+    ->setPatterns([
+        'slug' => '[a-z0-9-_]+'
```
Comment (ravage84) [inline en/development/routing.rst]: comma

File: en/development/routing.rst [hunk 2]
```diff
@@ -338,6 +396,44 @@ CakePHP, and should not be used unless you want the special meaning
...
+        'lang' => 'en|fr|es'
```
Comment (ravage84) [inline en/development/routing.rst]: comma

File: en/development/routing.rst [hunk 3]
```diff
@@ -356,16 +452,16 @@ functions::
...
-        );
+            ['controller' => 'Blogs', 'action' => 'view']
+        )
+        ->setPatterns([
+            // Define the route elements in the route template
+            // to pass as function arguments. Order matters since this
+            // will simply map ":id" to $articleId in your action
+            'pass' => ['id', 'slug'],
+            // Define a pattern that `id` must match.
+            'id' => '[0-9]+'
```
Comment (ravage84) [inline en/development/routing.rst]: comma

File: en/development/routing.rst [hunk 4]
```diff
@@ -631,19 +734,36 @@ with the following router connection::
...
         // This route only matches on POST requests.
-        $routes->connect(
+        $routes->post(
             '/reviews/start',
-            ['controller' => 'Reviews', 'action' => 'start', '_method' => 'POST']
+            ['controller' => 'Reviews', 'action' => 'start']
         );
+
+        // Match multiple verbs
+        // Prior to 3.5 use $options[_method] to set method
```
Comment (ravage84) [inline en/development/routing.rst]: ``$options['_method'] ``?

**Comment by markstory** [comment]
Thanks @ravage84

**Comment by ravage84** [comment]
Thanks!

**Pull request merged**